### PR TITLE
 Replace developer mode with test mode and send the event to Spoor with `is_live` set to false.

### PR DESCRIPTION
**Pull request by JakeChampion**
Chris Brown pointed out that `is_live` currently is not used to filter out test data but that it could be if o-tracking enforced a consistent setting, which is what o-tracking will now do.

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 0785e295dee8fd5bb490350188999a09f1076a2e
js: 4.27kb increase (1.33kb/gzip)

File: MIGRATION.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ The deprecated `user.passport_id` data property has been removed.
...
 ```
 
+`tracking.developer` has been removed, it has been replaced by the configuration parameter `test`. Set `test` to `true` to turn send events in test mode.
+```diff
+-tracking.developer(true);
++tracking.init({
+    test: true
+});
+```
+
```
Comment (JakeChampion) [inline MIGRATION.md:27]:
```suggestion


The configuration parameter `no_send` has been removed, it has been replaced by the configuration parameter `test`. Set `test` to `true` to turn send events in test mode.
```diff
-tracking.init({
-	no_send: true
-});
+tracking.init({
+    test: true
+});
```
```

File: MIGRATION.md [hunk 2]
```diff
@@ -17,6 +17,14 @@ The deprecated `user.passport_id` data property has been removed.
 +tracking.click.init();
 ```
 
+`tracking.developer` has been removed, it has been replaced by the configuration parameter `test`. Set `test` to `true` to turn send events in test mode.
+```diff
+-tracking.developer(true);
++tracking.init({
+    test: true
+});
```
Comment (JakeChampion) [inline MIGRATION.md:25]:
```suggestion
+    test: true
+});
```

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to e06cc883c8c22a65ec65cf1ae639c11f7c4d286d
js: 4.41kb increase (1.33kb/gzip)

File: MIGRATION.md [hunk 3]
```diff
@@ -19,6 +19,23 @@ The deprecated `user.passport_id` data property has been removed.
 +tracking.click.init();
 ```
 
+`tracking.developer` has been removed, it has been replaced by the configuration parameter `test`. Set `test` to `true` to turn send events in test mode.
```
Comment (notlee) [inline MIGRATION.md:22]: >to turn send events in test mode.

what now? also what does test mode do? send the event but the event is ignored or bucketed differently by Spoor?

File: MIGRATION.md [hunk 4]
```diff
@@ -19,6 +19,23 @@ The deprecated `user.passport_id` data property has been removed.
...
+```diff
+-tracking.developer(true);
++tracking.init({
++    test: true
++});
+```
+
+
+The configuration parameter `no_send` has been removed, it has been replaced by the configuration parameter `test`. Set `test` to `true` to turn send events in test mode.
```
Comment (notlee) [inline MIGRATION.md:31]: as above

File: test/main.test.js [hunk 5]
```diff
@@ -187,6 +186,76 @@ describe('main', function () {
 		const sent_data = callback.getCall(0).thisValue;
 
 		proclaim.equal(sent_data.system.environment, "prod");
+	});
+
+	it('should not allow system.is_live to be set on init', function () {
```
Comment (notlee) [inline test/main.test.js:191]: 👍

**Comment by notlee** [comment]
I love that you're taking the opportunity to make improvments as you go with feedback from the data team :)

File: MIGRATION.md (hunk 3 above)
Comment (JakeChampion) [reply MIGRATION.md:22]:
```suggestion
The `tracking.developer` function has been removed, it has been replaced by a configuration parameter named `test`. Set the `test` configuration parameter to `true` to have similar behavior to the old `developer` mode.
```

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 85580079c68f6cbe67efa89dac31f8c7b4121412
js: 4.41kb increase (1.33kb/gzip)

File: MIGRATION.md (hunk 4 above)
Comment (JakeChampion) [reply MIGRATION.md:31]:
```suggestion
The configuration parameter `no_send` has been removed, it has been replaced by a configuration parameter named `test`. The events will be sent but will be marked as test events, which Spoor will ignore.
```

Comment (notlee) [reply MIGRATION.md]: Nice!

File: MIGRATION.md (hunk 3 above)
Comment (notlee) [reply MIGRATION.md]: Nice!

File: README.md [hunk 6]
```diff
@@ -52,6 +52,22 @@ Here is the corresponding tracking pixel setup:
 
 ### Tracking with JavaScript
 
+#### Developer/Test Mode
+
+O-tracking has a development/test mode which will mark the events as test events, and also write to the console extra debugging information.
```
Comment (notlee) [inline README.md:57]:
```suggestion
o-tracking has a development/test mode which will mark the events as test events, and also write to the console extra debugging information.
```

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 3a3538651dc27f128a5614efb990387958c1b2e4
js: 4.41kb increase (1.33kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to 8511bb79d15d5dc33e52297bc7b2d08321a64aec
js: 4.41kb increase (1.33kb/gzip)

**Comment by origamiserviceuser** [comment]
o-tracking bundle size difference from 2.0.10 to bee7d0990259a7e5a8e2294ed41c3744ee8f3271
js: 4.41kb increase (1.33kb/gzip)

**Pull request merged**